src/main/scala/org/ergoplatform/ErgoBoxCandidate.scala

```diff
@@ -122,10 +122,12 @@ object ErgoBoxCandidate {
     }
 
     def parseBodyWithIndexedDigests(digestsInTx: Option[Coll[TokenId]], r: SigmaByteReader): ErgoBoxCandidate = {
+      val previousPositionLimit = r.positionLimit
+      r.positionLimit = r.position + ErgoBox.MaxBoxSize
       val value = r.getULong()
       val tree = ErgoTreeSerializer.DefaultSerializer.deserializeErgoTree(r)
       val creationHeight = r.getUInt().toInt
-      val addTokensCount = r.getByte()
```

Comment:
Looks like this breaks binary format?

Reply:
@catena2w I don't think so. We're writing unsigned values, but we were reading signed values. I just fixed reading.

Reply:
nope `@inline override def getUByte(): Int = getByte() & 0xFF`